### PR TITLE
feat: Add SLES subscription mounts

### PR DESCRIPTION
**Pull request by rollandf**
Add required mounts for MOFED in case of SLES.

### CODE REVIEW

File: pkg/state/state_ofed.go [hunk 1]
```diff
@@ -829,8 +845,8 @@ func (s *stateOFED) handleCertConfig(
 func (s *stateOFED) handleSubscriptionVolumes(
 	ctx context.Context, osname string, runtime string, mounts *additionalVolumeMounts) error {
 	reqLogger := log.FromContext(ctx)
-	if osname == "rhel" && runtime != nodeinfo.CRIO {
-		reqLogger.V(consts.LogLevelDebug).Info("Setting subscription mounts for RHEL with non CRIO container runtime")
+	if osname == "rhel" && runtime != nodeinfo.CRIO || osname == "sles" {
```
Comment (heyvister1) [inline pkg/state/state_ofed.go:848]: .nit For code readability improvement
```suggestion
	if (osname == "rhel" && runtime != nodeinfo.CRIO) || osname == "sles" {
```

File: pkg/state/state_ofed.go [hunk 2]
```diff
@@ -111,13 +111,15 @@ var CertConfigPathMap = map[string]string{
...
 	"rhel":   "/etc/pki/ca-trust/extracted/pem",
+	"sles":   "/etc/ssl",
 }
 
 // RepoConfigPathMap indicates standard OS specific paths for repository configuration files
 var RepoConfigPathMap = map[string]string{
 	"ubuntu": "/etc/apt/sources.list.d",
 	"rhcos":  "/etc/yum.repos.d",
 	"rhel":   "/etc/yum.repos.d",
+	"sles":   "/etc/zypp/repos.d/",
```
[the remark this answers is not in the document]
Comment (rollandf) [reply pkg/state/state_ofed.go]: Done

File: pkg/state/state_ofed.go (hunk 1 above)
Comment (rollandf) [reply pkg/state/state_ofed.go]: Done

**Review by heyvister1 — APPROVED** [review]
LGTM

File: pkg/state/state_ofed_test.go [hunk 3]
```diff
@@ -375,11 +375,63 @@ var _ = Describe("MOFED state test", func() {
...
+		cr.Name = "nic-cluster-policy"
+		cr.Spec.OFEDDriver = &v1alpha1.OFEDDriverSpec{
+			ImageSpec: v1alpha1.ImageSpec{
+				Image:      "mofed",
+				Repository: "nvcr.io/mellanox",
+				Version:    "23.10-0.5.5.0",
+			},
+		}
+
+		By("Creating NodeProvider with 1 Nodes, SLES")
```
Comment (e0ne) [inline pkg/state/state_ofed_test.go:409]: @rollandf thanks for the adding unit-tests too!

**Review by e0ne — APPROVED** [review]

**Pull request merged**